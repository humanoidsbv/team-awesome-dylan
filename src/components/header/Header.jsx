import React from 'react';

import styles from './Header.module.css';
import Logo from '../logo/Logo';
import Navigation from '../navigation/Navigation';
import MenuToggle from '../menu-toggle/MenuToggle';
import ProfileButton from '../profile-button/ProfileButton';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isMenuVisible: false };
    this.toggleMenu = this.toggleMenu.bind(this);
  }

  toggleMenu() {
    const { isMenuVisible } = this.state;
    this.setState({ isMenuVisible: !isMenuVisible });
  }

  render() {
    const { isMenuVisible } = this.state;
    return (
      <header
        className={`${styles.header} ${
          isMenuVisible ? styles.headerMenuVisible : ''
        }`}
      >
        <Logo />
        <Navigation isMenuVisible={isMenuVisible} />
        <MenuToggle
          isMenuVisible={isMenuVisible}
          toggleMenu={this.toggleMenu}
        />
        <ProfileButton />
      </header>
    );
  }
}

export default Header;
