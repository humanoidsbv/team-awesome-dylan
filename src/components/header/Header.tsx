import React, { useState } from 'react';

import Logo from '../logo/Logo';
import MenuToggle from '../menu-toggle/MenuToggle';
import Navigation from '../navigation/Navigation';
import ProfileButton from '../profile-button/ProfileButton';
import styles from './Header.module.css';

function Header(): React.ReactElement {
  const [isMenuVisible, setMenuVisibility] = useState(false);

  const toggleMenu = (): void => {
    setMenuVisibility(!isMenuVisible);
  };

  return (
    <header
      className={`${styles.header} ${
        isMenuVisible ? styles.headerMenuVisible : ''
      }`}
    >
      <Logo />
      <Navigation isMenuVisible={isMenuVisible} />
      <MenuToggle isMenuVisible={isMenuVisible} toggleMenu={toggleMenu} />
      <ProfileButton />
    </header>
  );
}

export default Header;
