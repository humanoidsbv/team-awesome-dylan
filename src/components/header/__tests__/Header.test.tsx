import * as React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

it('matches snapshot', () => {
  const header = shallow(
    <Header />
  );
  expect(header).toMatchSnapshot();
});
