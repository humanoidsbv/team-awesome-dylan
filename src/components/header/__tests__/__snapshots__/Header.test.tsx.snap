// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header "
>
  <Logo />
  <Navigation
    isMenuVisible={false}
  />
  <MenuToggle
    isMenuVisible={false}
    toggleMenu={[Function]}
  />
  <ProfileButton />
</header>
`;
